File: DataStructures/BST/set1_2.py
class Node:
    def __init__(self, key):
        self.left = None
        self.right = None
        self.val = key

    def insert(root, key):
        if root is None:
            return Node(key)
        elif root.val < key:
            root.right = insert(root.right, key)
        else:
            root.left = insert(root.left, key)
        return root
    
    def inOrder(root):
        if root:
            inOrder(root.left)
            print(root.val)
            inOrder(root.right)

if __name__ == "__main__":
    r = Node(50)
    r = insert(r, 30)
    r = insert(r, 40)
    r = insert(r, 50)
    r = insert(r, 60)
    r = insert(r, 70)
    r = insert(r, 80)
    inOrder(r)

File: DataStructures/BST/set1_2.cs
using System;

class BST{

    public class Node
    {
        public int key, 
        public Node left, right;

        public Node(int item)
        {
            key = item;
            left = right = null;
        }
    }

    Node root;

    BST()
    {
        root = null;
    }

    void insert(int key)
    {
        root = insertRec(root, key);
    }

    Node insertRec(root, key)
    {
        if(root == null)
        {
            root = new Node(key);
            return root;
        }

        if(root.key < key)
        {
            root.right = insertRec(root.right, key);
        }else if (root.key > key)
        {
            root.left = insertRec(root.left, key);
        }
    }

    void inorder()
    {
        inorderRec(root);
    }

    void inorderRec(root)
    {
        if(root != null)
        {
            inorderRec(root.left);
            Conosle.Write(root.key);
            inorderRec(root.right);
        }
    }
}

public static void Main(String[] args)
{
    BST tree = new BST();
    tree.insert(20);
    tree.insert(40);
    tree.insert(50);
    tree.insert(60);
    tree.insert(70);
    tree.insert(80);
    tree.insert(90);

    tree.inorder();
}